resource-server/src/main/java/com/okta/spring/example/ResourceServerExampleApplication.java

```diff
@@ -50,7 +48,7 @@ protected void configure(HttpSecurity http) throws Exception {
 
         @GetMapping("/api/userProfile")
         @PreAuthorize("hasAuthority('SCOPE_profile')")
-        public Map<String, Object> getUserDetails(JwtAuthenticationToken authentication) {
+        public Map<String, Object> getUserDetails(AbstractOAuth2TokenAuthenticationToken authentication) {
```

Comment:
`AbstractOAuth2TokenAuthenticationToken` is the abstraction for `JwtAuthenticationToken` (JWT) & `BearerTokenAuthentication` (Opaque Token).